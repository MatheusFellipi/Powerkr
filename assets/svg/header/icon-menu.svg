<svg width="36" height="36" viewBox="0 0 36 36" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M11.125 17.4375H24.875C25.0242 17.4375 25.1673 17.4968 25.2727 17.6023C25.3782 17.7077 25.4375 17.8508 25.4375 18C25.4375 18.1492 25.3782 18.2923 25.2727 18.3977C25.1673 18.5032 25.0242 18.5625 24.875 18.5625H11.125C10.9758 18.5625 10.8327 18.5032 10.7273 18.3977C10.6218 18.2923 10.5625 18.1492 10.5625 18C10.5625 17.8508 10.6218 17.7077 10.7273 17.6023C10.8327 17.4968 10.9758 17.4375 11.125 17.4375Z" fill="#CDD5DF" stroke="#CDD5DF" stroke-width="0.125"/>
  <path d="M24.875 13.5625H11.125C10.9758 13.5625 10.8327 13.5032 10.7273 13.3977C10.6218 13.2923 10.5625 13.1492 10.5625 13C10.5625 12.8508 10.6218 12.7077 10.7273 12.6023C10.8327 12.4968 10.9758 12.4375 11.125 12.4375H24.875C25.0242 12.4375 25.1673 12.4968 25.2727 12.6023C25.3782 12.7077 25.4375 12.8508 25.4375 13C25.4375 13.1492 25.3782 13.2923 25.2727 13.3977C25.1673 13.5032 25.0242 13.5625 24.875 13.5625Z" fill="#CDD5DF" stroke="#CDD5DF" stroke-width="0.125"/>
  <path d="M24.875 22.375H11.125C10.9592 22.375 10.8003 22.4408 10.6831 22.5581C10.5658 22.6753 10.5 22.8342 10.5 23C10.5 23.1658 10.5658 23.3247 10.6831 23.4419C10.8003 23.5592 10.9592 23.625 11.125 23.625H24.875C25.0408 23.625 25.1997 23.5592 25.3169 23.4419C25.4342 23.3247 25.5 23.1658 25.5 23C25.5 22.8342 25.4342 22.6753 25.3169 22.5581C25.1997 22.4408 25.0408 22.375 24.875 22.375Z" fill="#CDD5DF"/>
  </svg>
  