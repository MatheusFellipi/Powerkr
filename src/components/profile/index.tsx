import { Avatar, Text, View } from "tamagui";

export default function Home() {
  return (
    <View>
      <Avatar
        circular
        size="$15"
      >
        <Avatar.Image
          accessibilityLabel="Cam"
          src="https://images.unsplash.com/photo-1548142813-c348350df52b?&w=150&h=150&dpr=2&q=80"
        />
        <Avatar.Fallback backgroundColor="$blue10" />
      </Avatar>
      <Text>Colaborador nome</Text>
      <Text>Departamento</Text>
    </View>
  );
}
