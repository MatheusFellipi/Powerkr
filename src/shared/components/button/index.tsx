import { Button, styled } from "tamagui";

export const BtnComponent = styled(Button, {
  fontSize: 14,
  borderRadius: 24,
  fontFamily: "InterBold",
  borderColor: "#4B5565",
  bg: "#202939",
  variants: {
    tabs: {
      activated: {},
      disabled: {
        borderColor: "$colorTransparent",
        bg: "$colorTransparent"
      }
    },
    styles: {
      login: {
        bg: "#F7B519",
        color: "#000"
      }
    }
  } as const
});
