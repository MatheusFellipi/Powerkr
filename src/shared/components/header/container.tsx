import { ReactNode } from "react";
import { View } from "tamagui";

type Props = {
  children: ReactNode;
};

export function Container({ children }: Props) {
  return (
    <View
      fd="row"
      justifyContent="center"
      position="relative"
    >
      {children}
    </View>
  );
}
