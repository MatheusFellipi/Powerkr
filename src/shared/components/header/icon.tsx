import { SvgProps } from "react-native-svg";
import { View } from "tamagui";

type Props = {
  Icon: React.FC<SvgProps>;
  rightIcon?: boolean;
};

export function Icon({ Icon, rightIcon = false }: Props) {
  return (
    <View
      position="absolute"
      {...(rightIcon ? { right: "$0" } : { left: "$0" })}
    >
      <Icon />
    </View>
  );
}
