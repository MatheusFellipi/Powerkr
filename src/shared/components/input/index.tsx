import { useState } from "react";
import { Pressable, TextInputProps } from "react-native";
import { Icons } from "@assets/index";
import { Input, InputProps, Label, View } from "tamagui";

type Props = { label: string } & InputProps;

export const InputComponent = ({
  label,
  id,
  onChangeText,
  secureTextEntry,
  ...props
}: Props) => {
  const [showPassword, setShowPassword] = useState(secureTextEntry);
  return (
    <View>
      <Label
        color={"#CDD5DF"}
        htmlFor={id}
      >
        {label}
      </Label>
      <View
        fd="row"
        jc="space-between"
        ai="center"
        borderRadius={24}
        bg="#364152"
        borderWidth={1}
        borderColor="#4B5565"
      >
        <Input
          {...props}
          width={"90%"}
          borderRadius={24}
          secureTextEntry={showPassword}
          bg="$backgroundTransparent"
          borderColor="$backgroundTransparent"
        />
        {secureTextEntry && (
          <Pressable
            onPress={() => setShowPassword(!showPassword)}
            style={{
              width: "10%",
              paddingRight: 10
            }}
          >
            {showPassword ? (
              <Icons.EyeOn
                fill={"#A0AEC0"}
                height={25}
                width={25}
              />
            ) : (
              <Icons.EyeOff
                fill={"#A0AEC0"}
                height={25}
                width={25}
              />
            )}
          </Pressable>
        )}
      </View>
    </View>
  );
};
