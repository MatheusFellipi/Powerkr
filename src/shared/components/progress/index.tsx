import { CircularProgressComponent } from "@shared/components/progress/circular";

export const Progress = {
  Circular: CircularProgressComponent
};
