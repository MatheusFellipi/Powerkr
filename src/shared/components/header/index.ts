import { Container } from "@shared/components/header/container";
import { Icon } from "@shared/components/header/icon";
import { TextComponent } from "@shared/components/textComponent";

export const HeaderComponent = {
  Text: TextComponent,
  Container: Container,
  Icon: Icon
};
