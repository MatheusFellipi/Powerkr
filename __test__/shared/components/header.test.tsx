import { HeaderComponent } from "@shared/components/header";
import { render } from "@testing-library/react-native";
import { TamaguiProvider } from "tamagui";

import config from "../../../tamagui.config";

describe("<App />", () => {
  it("renders correctly", () => {
    const tree = render(
      <TamaguiProvider config={config}>
        <HeaderComponent.Container>
          <HeaderComponent.Text pin="description">test</HeaderComponent.Text>
        </HeaderComponent.Container>
      </TamaguiProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
