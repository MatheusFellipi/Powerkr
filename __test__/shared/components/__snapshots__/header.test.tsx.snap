// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "center",
      "position": "relative",
    }
  }
>
  <Text
    fontFamily="Inter"
    style={
      {
        "color": "#9AA4B2",
        "fontFamily": "Inter",
        "fontSize": 14,
      }
    }
    suppressHighlighting={true}
  >
    test
  </Text>
</View>
`;
